pkg/platform/kube/client/deployer.go

```diff
@@ -146,6 +160,278 @@ func (d *Deployer) Deploy(ctx context.Context,
+
+	// get existing function secret
+	d.logger.DebugWithCtx(ctx, "Getting function secret", "functionName", functionName, "functionNamespace", functionNamespace)
+	functionSecretData, err := d.getFunctionSecretData(ctx, functionName, functionNamespace)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get function secret")
+	}
+
+	// if secret exists, get the data
+	if functionSecretData != nil {
+		functionSecretMap, err := functionconfig.DecodeSecretData(functionSecretData)
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to decode function secret data")
+		}
+		return functionSecretMap, nil
+	}
+
+	// secret doesn't exist
+	d.logger.DebugWithCtx(ctx, "Function secret doesn't exist", "functionName", functionName, "functionNamespace", functionNamespace)
+	return nil, nil
+}
+
+func (d *Deployer) getFunctionSecretData(ctx context.Context, functionName, functionNamespace string) (map[string][]byte, error) {
+
+	// get existing function secret
+	functionSecrets, err := d.consumer.KubeClientSet.CoreV1().Secrets(functionNamespace).List(ctx, metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", common.NuclioResourceLabelKeyFunctionName, functionName),
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get function secret")
+	}
+
+	// if secret exists, get the data
+	for _, functionSecret := range functionSecrets.Items {
+
+		// if it is a flex volume secret, skip it
+		if strings.HasPrefix(functionSecret.Name, functionconfig.NuclioFlexVolumeSecretNamePrefix) {
+			continue
+		}
+
+		return functionSecret.Data, nil
+	}
+
+	return nil, nil
+}
+
+func (d *Deployer) createOrUpdateFunctionSecret(ctx context.Context,
+	encodedSecretsMap map[string]string,
+	name,
+	namespace string) error {
+
+	secretConfig := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: d.generateFunctionSecretName(name),
+			Labels: map[string]string{
+				common.NuclioResourceLabelKeyFunctionName: name,
+			},
+		},
+		Type:       functionconfig.NuclioSecretType,
+		StringData: encodedSecretsMap,
+	}
+
+	if len(encodedSecretsMap) > 0 {
+		if err := d.createOrUpdateSecret(ctx, namespace, secretConfig); err != nil {
+			return errors.Wrap(err, "Failed to create function secret")
+		}
+		return nil
+	}
+
+	// if secret exists and there are no secrets to set, delete the secret
+	return d.deleteExistingSecret(ctx, namespace, secretConfig.Name)
+}
+
+func (d *Deployer) generateFunctionSecretName(functionName string) string {
+	secretName := fmt.Sprintf("%s%s", functionconfig.NuclioSecretNamePrefix, functionName)
+	secretNameLimit := 63 - (len(functionconfig.NuclioSecretNamePrefix) + len(functionName) + 2)
+	if len(secretName) > secretNameLimit {
+		secretName = secretName[:secretNameLimit]
+	}
+	return secretName
+}
+
+func (d *Deployer) handleFlexVolumeSecret(ctx context.Context, functionName, functionNamespace string, secretsMap map[string]string) error {
```

Comment:
handle is a bit of a generic name. do you always create/delete secrets? will you reuse if access key was not changed?

Reply:
see comment above